source/dubregistry/registry.d

```diff
@@ -408,8 +409,31 @@ class DubRegistry {
 		dbver.info = info.info;
 
 		try {
-			rep.readFile(reference.sha, InetPath("/README.md"), (scope input) { dbver.readme = input.readAllUTF8(); });
-		} catch (Exception e) { logDiagnostic("No README.md found for %s %s", dbpack.name, ver); }
+			auto files = rep.listFiles(reference.sha, InetPath("/"));
+			// check exactly for readme.me
+			ptrdiff_t readme;
+			readme = files.countUntil!(a => a.type == RepositoryFile.Type.file && a.path.head.name.asUpperCase.equal("README.MD"));
+			if (readme == -1) {
+				// check exactly for readme
+				readme = files.countUntil!(a => a.type == RepositoryFile.Type.file && a.path.head.name.asUpperCase.equal("README"));
+			}
+			if (readme == -1) {
+				// check for all other readmes such as README.txt, README.jp.md, etc.
+				readme = files.countUntil!(a => a.type == RepositoryFile.Type.file && a.path.head.name.asUpperCase.startsWith("README"));
+			}
+
+			if (readme != -1) {
+				rep.readFile(reference.sha, files[readme].path, (scope input) {
+					dbver.readme = input.readAllUTF8();
+					string ext = files[readme].path.head.name;
+					// endsWith doesn't like to work with asLowerCase
+					dbver.readmeMarkdown = ext.endsWith(".md") || ext.endsWith(".MD") || ext.endsWith(".Md") || ext.endsWith(".mD");
```

Comment:
Yeah the problems is that `asLowerCase` doesn't yield a bidirectional range :/
A workaround is this:

```
"foo.mD".retro.asLowerCase.startsWith(".md".retro)
````

https://run.dlang.io/is/PAuvw4

Though it's not very nice either.
Another one:

```
alias lowerCasePred = (a, b) => a.only.sicmp(b.only) == 0;
writeln("foo.mD".endsWith!lowerCasePred(".md"));
```

https://run.dlang.io/is/gdhzFQ

Reply:
uh any problem with this? It's just 4 checks

Reply:
Or just `ext.extension.icmp(".md") == 0`.

Reply:
well probably not icmp though, we don't want combined unicode characters in the filenames